Makefile

```diff
@@ -15,47 +14,45 @@ env: $(VENV)
 clean-env:
 	rm -rf $(VENV)
 
-.PHONY: install-hooks
-install-hooks: $(VENV) .pre-commit-config.yaml
+$(HOOKS): $(VENV) .pre-commit-config.yaml
 	$(VENV)/bin/pre-commit install -f --install-hooks
-	cargo +nightly fmt --help > /dev/null || rustup component add rustfmt-preview --toolchain nightly
-	cargo +nightly clippy --help > /dev/null || cargo +nightly install clippy
+	cargo fmt --help > /dev/null || rustup component add rustfmt
+	cargo clippy --help > /dev/null || rustup component add clippy
 
+.PHONY: install-hooks
+install-hooks: $(HOOKS)
 
 .PHONY: clean-hooks
 clean-hooks:
 	rm -rf $(HOOKS)
 
-
 # LINTING
 
 .PHONY: lint
 lint:
-	 cargo +nightly fmt
+	 cargo fmt
 
 .PHONY: clean-lint
 clean-lint:
 	find . -type f -name *.rs.bk -delete
 
 .PHONY: clippy
 clippy:
-	cargo +nightly clippy --all-features
-
+	cargo clippy --all-features
 
 # TESTING
 
 .PHONY: test
-test:
+test: install-hooks
 	cargo test --all-features
-
+	$(VENV)/bin/pre-commit run --all-files
```

Comment:
Not sure I approve of `make test` having side effects (namely changing code with `fmt`), but that's not a huge deal :)

Reply:
Yeah, but how do we make sure people are formatting their code? Not everyone enables pre-commits.